### DISCUSSION TITLE
How to add custom commands to Surf

### DISCUSSION THREAD

**Discussion by erik-zicht** [TYPO3/Surf#607] [Ideas]
Currently, the available commands in Surf are:

```bash
Available commands:
  deploy    Describes the flow for the given name
  describe  Describes the flow for the given name
  help      Display help for a command
  list      List commands
  migrate   Migrates old deployment definitions to new Surf version
  rollback  Rollback current to previous release and remove current folder
  show      Shows all the deployments depending on the directory configuration
  simulate  Simulates the deployment for the given name
```

What if I want to add my own commands? E.g. `vendor/bin/surf ssh prod` to connect to a server. Or `vendor/bin/surf db:dump prod` to fetch a database-dump from remote.

Is that currently possible, or does Surf need internal refactoring to be able to add new custom commands?

**Comment by simonschaufi** [comment]
You better use TYPO3 Console for that or your own symfony console application. There you can add any command you want because Surf does not support custom commands. Surf is just a symfony console application.
